lib/util/http/host-rules.ts

```diff
@@ -120,10 +121,23 @@ export function applyHostRules(url: string, inOptions: GotOptions): GotOptions {
   return options;
 }
 
-export function getRequestLimit(url: string): number | null {
-  const hostRule = hostRules.find({
-    url,
-  });
-  const limit = hostRule.concurrentRequestLimit;
-  return typeof limit === 'number' && limit > 0 ? limit : null;
+export function getConcurrentRequestsLimit(url: string): number | null {
+  const hostRule = hostRules.find({ url });
+  const concurrentLimit = hostRule.concurrentRequestLimit;
+  return typeof concurrentLimit === 'number' && concurrentLimit > 0
+    ? concurrentLimit
+    : null;
+}
+
+export function getThrottleOptions(
+  url: string
+): { limit: number; interval: number } | null {
+  const hostRule = hostRules.find({ url });
+  const throttleLimit = hostRule.maxRequestsPerSecond;
+  if (!is.number(throttleLimit)) {
+    return null;
+  }
+  const limit = Math.floor(throttleLimit * 1000);
+  const interval = 1000 * 1000;
```

Comment:
I think the term "interval" usually means the gap between things, whereas here you seem to be using it as the time window, so I think terminology should be changed.

If we were to implement this simply, meaning "enforce the gap between each request" then it would be simpler, a well as more likely to satisfy other parties. e.g. if a party says maximum 10 RPS but we send 10 in the first millisecond, then wait 999ms, then send 10, etc - they likely won't be happy.

So what if you *actually* return the interval? e.g. if 1 RPS the interval is 1000ms, if 10RPS then 100ms, if it's 0.5RPS then the interval is 2000ms, etc.

Reply:
I think I finally implemented your idea